app/models/concerns/interlocks.rb

```diff
@@ -1,53 +1,77 @@
 module Interlocks
   # extend ActiveSupport::Concern
   def interlocks(page = 1)
-    # TODO: create position_scope for Link
-    org_ids = connecting_ids(rel_cat_id)
-    people_and_org_ids = paginate(page,
-                                  Entity::PER_PAGE,
-                                  connected_id_hashes_for(org_ids, rel_cat_id))
-
-    entities_by_id = Entity.lookup_table_for(collapse(people_and_org_ids))
-
-    people_and_org_ids.map do |connected_id_hash|
+    id_hashes = paginate(page, Entity::PER_PAGE, connected_id_hashes)
+    entities_by_id = Entity.lookup_table_for(collapse(id_hashes))
+    id_hashes.map do |id_hash|
       {
-        "person" => entities_by_id.fetch(connected_id_hash[:connected_id]),
-        "orgs" => connected_id_hash[:connecting_ids].map { |id| entities_by_id.fetch(id) }
+        "connected_entity" => entities_by_id.fetch(id_hash[:connected_id]),
+        "connecting_entities" => id_hash[:connecting_ids].map { |id| entities_by_id.fetch(id) }
       }
     end
   end
 
   private
 
-  def rel_cat_id
-    Relationship::POSITION_CATEGORY
-  end
-
-  def connecting_ids(cat_id)
-    links
-      .where(entity1_id: id,
-             category_id: cat_id,
-             is_reverse: false)
-      .pluck(:entity2_id)
+  def connecting_ids
+    relationships
```

Comment:
do note that by doing entity#relationships, [it uses the link table via a join](https://github.com/public-accountability/littlesis-rails/blob/master/app/models/entity.rb): 

``` rb
has_many :links, foreign_key: "entity1_id", inverse_of: :entity, dependent: :destroy
has_many :relationships, through: :links
```

if you don't need information from the relationship table it's usually slightly more performat to just deal with the links table. Was there a particular reason to  switch to using relationships here?

Reply:
cool, thanks for pointing that out! i'm not attached to relationship instead of link and will happily replace it on the next round of refactors for the next PR.

the only reason i replaced it was that it seemed slightly logically simpler to reason about relationships than a "link where `is_reverse` is not true" (which is just a relationship). i'd also guessed that since the links table has (i think?) twice the number of rows as the links table, that querying the relationships table would be marginally faster. i had no idea that querying relatinoships joined links. setting aside for today why it does that (if links are derrived from relationships, why would relationships need to depend on links? wouldn't you expect it to be the other way around?) as a small fish in the bigger-fish-to-fry ocean of incidental complexity in which we are swimming our way to simplicity, i'm happy to swim with the current and use links.

this might be a good time to point out that as i start to familiarize myself with the graph theory used in social network analysis (see [this book](https://www.amazon.com/dp/1473952123) for starters, more links in the wiki on the way!), i'm starting to accumulate a list of ways in which i think our modeling of relationships/links might be a bit at odds with the graph operations we might want to perform on them... (see: e.g.: the "directedness" of employment edges. why are they this way? need they be so? does it model anything useful in the domain to have them be directed? how does it affect our ability to traverse the graph to make it this way? -- keeping in mind that you may not traverse an edge in the opposite direction of its directedness...) but that's a much bigger discussion -- one that fortunately we have a few more months to dig into in our spare time.

speaking of which: let me stop writing this comment and get back to coding!!! 🤣